### ISSUE TITLE
Python Diwali Sales codes

### ISSUE THREAD

**Issue by PrachiSinghai1105** [PrachiSinghai1105/Superstore#3]
# import python libraries

import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt # visualizing data
%matplotlib inline
import seaborn as sns

# import csv file
df = pd.read_csv('Diwali Sales Data.csv', encoding= 'unicode_escape')

df.shape

df.head(10)

df.info()

#drop unrelated/blank columns
df.drop(['Status', 'unnamed1'], axis=1, inplace=True)

#check for null values
pd.isnull(df).sum()

# drop null values
df.dropna(inplace=True)

# change data type
df['Amount'] = df['Amount'].astype('int')

df['Amount'].dtypes

df.columns

#rename column
df.rename(columns= {'Marital_Status':'Shaadi'})

# describe() method returns description of the data in the DataFrame (i.e. count, mean, std, etc)
df.describe()

# use describe() for specific columns
df[['Age', 'Orders', 'Amount']].describe()

#Exploratory Data Analysis
#Gender
# plotting a bar chart for Gender and it's count

ax = sns.countplot(x = 'Gender',data = df)

for bars in ax.containers:
    ax.bar_label(bars)

# plotting a bar chart for gender vs total amount

sales_gen = df.groupby(['Gender'], as_index=False)['Amount'].sum().sort_values(by='Amount', ascending=False)

sns.barplot(x = 'Gender',y= 'Amount' ,data = sales_gen)

#Age

ax = sns.countplot(data = df, x = 'Age Group', hue = 'Gender')

for bars in ax.containers:
    ax.bar_label(bars)

# Total Amount vs Age Group
sales_age = df.groupby(['Age Group'], as_index=False)['Amount'].sum().sort_values(by='Amount', ascending=False)

sns.barplot(x = 'Age Group',y= 'Amount' ,data = sales_age)

#State

# total number of orders from top 10 states

sales_state = df.groupby(['State'], as_index=False)['Orders'].sum().sort_values(by='Orders', ascending=False).head(10)

sns.set(rc={'figure.figsize':(15,5)})
sns.barplot(data = sales_state, x = 'State',y= 'Orders')

# total amount/sales from top 10 states

sales_state = df.groupby(['State'], as_index=False)['Amount'].sum().sort_values(by='Amount', ascending=False).head(10)

sns.set(rc={'figure.figsize':(15,5)})
sns.barplot(data = sales_state, x = 'State',y= 'Amount')

#Marital Status

ax = sns.countplot(data = df, x = 'Marital_Status')

sns.set(rc={'figure.figsize':(7,5)})
for bars in ax.containers:
    ax.bar_label(bars)

sales_state = df.groupby(['Marital_Status', 'Gender'], as_index=False)['Amount'].sum().sort_values(by='Amount', ascending=False)

sns.set(rc={'figure.figsize':(6,5)})
sns.barplot(data = sales_state, x = 'Marital_Status',y= 'Amount', hue='Gender')

#Occupation

sns.set(rc={'figure.figsize':(20,5)})
ax = sns.countplot(data = df, x = 'Occupation')

for bars in ax.containers:
    ax.bar_label(bars)

#Product Category

sns.set(rc={'figure.figsize':(20,5)})
ax = sns.countplot(data = df, x = 'Occupation')

for bars in ax.containers:
    ax.bar_label(bars)

sales_state = df.groupby(['Product_Category'], as_index=False)['Amount'].sum().sort_values(by='Amount', ascending=False).head(10)

sns.set(rc={'figure.figsize':(20,5)})
sns.barplot(data = sales_state, x = 'Product_Category',y= 'Amount')

sales_state = df.groupby(['Product_ID'], as_index=False)['Orders'].sum().sort_values(by='Orders', ascending=False).head(10)

sns.set(rc={'figure.figsize':(20,5)})
sns.barplot(data = sales_state, x = 'Product_ID',y= 'Orders')

# top 10 most sold products (same thing as above)

fig1, ax1 = plt.subplots(figsize=(12,7))
df.groupby('Product_ID')['Orders'].sum().nlargest(10).sort_values(ascending=False).plot(kind='bar')

**Comment by muha99** [comment]
hey guys i don't understand anything in this website  and i have a problem with python code i have created to predict client churn , means to predict if a client is gonna leave or stay as a client here is the code i have created a import csv
import pandas as pd
import numpy as np
from sklearn.feature_extraction.text import TfidfVectorizer
import joblib  
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score
from sklearn.feature_extraction.text import TfidfVectorizer

# Define the column names and data separately
column_names = ["numero-du-client","age", "duree-dengagement-en-semaine", "nombre-de-transactions","le-prix-d-achat-le-plus-eleve-en-euro", "plaintes-precedents", "commentaires", "churn"]
data = [
    (1000, 23, 8, 5, 300, 0, "les meilleurs produits", 0),
    (1001, 18, 3, 8, 129, 0, "tres bien", 0),
    (1002, 25, 1.5, 2, 30, 2, "tres mauvaise service", 1),
    (1003, 30, 12, 4, 15, 1, "aucun commentaire", 1),
    (1004, 29, 48, 20, 430, 3, "moyen", 0),
    (1005, 45, 12, 3, 34, 0, "tres chere", 1),
    (1006, 67, 43, 55, 333, 1, "j-adore", 0),
    (1007, 33, 23, 2, 76, 2, "pas mal", 0),
    (1008, 22, 10, 4, 210, 1, "plutot satisfaisant", 1),
    (1009, 28, 15, 6, 89, 0, "excellent service", 0),
    (1010, 50, 5, 8, 120, 0, "assez bien", 1),
    (1011, 36, 25, 15, 450, 2, "ameliorer la rapidite", 1),
    (1012, 31, 3, 7, 80, 1, "service tres lent", 1),
    (1013, 63, 32, 12, 750, 3, "bon rapport qualite-prix", 0),
    (1014, 20, 2, 4, 60, 1, "rien a signaler", 0),
    (1015, 55, 38, 18, 600, 2, "tres satisfait", 0),
    (1016, 40, 18, 10, 350, 1, "moyen", 1),
    (1017, 42, 10, 5, 110, 1, "peu mieux faire", 1),
    (1018, 47, 20, 14, 500, 0, "top service", 0),
    (1019, 52, 29, 20, 800, 3, "tres bien", 0),
    (1020, 58, 12, 6, 180, 2, "service rapide", 0),
    (1021, 48, 5, 7, 160, 2, "moyenne qualite", 1),
    (1022, 41, 18, 9, 220, 0, "plutot cher", 1),
    (1023, 38, 23, 13, 360, 1, "a eviter", 1),
    (1024, 53, 14, 11, 400, 0, "vaut le coup", 0),
    (1025, 27, 8, 6, 250, 1, "satisfait", 0),
    (1026, 32, 11, 4, 70, 0, "rapport qualite-prix excellent", 0),
    (1027, 26, 15, 12, 310, 2, "service lent", 1),
    (1028, 35, 9, 10, 90, 0, "tres bien pour le prix", 1),
    (1029, 37, 18, 5, 180, 1, "meilleur service jamais eu", 0),
    (1030, 49, 7, 6, 110, 2, "a deconseiller", 1),
    (1031, 54, 20, 8, 200, 1, "pas cher", 0),
    (1032, 59, 30, 20, 700, 0, "tres satisfaisant", 0),
    (1033, 30, 6, 5, 70, 1, "moyen", 1),
    (1034, 43, 19, 14, 360, 3, "qualite superbe", 0),
    (1035, 34, 7, 4, 60, 0, "rien a dire", 0),
    (1036, 46, 26, 18, 600, 1, "tout bon", 0),
    (1037, 60, 21, 8, 250, 2, "service rapide", 0),
    (1038, 39, 10, 7, 170, 1, "pas mal", 0),
    (1039, 55, 15, 11, 450, 2, "ameliorez la rapidite", 1),
]

# Write the data to a CSV file
with open("data.csv", "w", newline='') as clientsdata:
    writer = csv.writer(clientsdata)
    writer.writerow(column_names)  # Write column names
    writer.writerows(data)  # Write data rows

# Load the data
data = pd.read_csv('data.csv')  # Remove encoding='ISO-8859-1'

# Convert TF-IDF features for the "commentaires" column
tfidf_vectorizer = TfidfVectorizer()
tfidf_features = tfidf_vectorizer.fit_transform(data['commentaires']).toarray()

# Combine the TF-IDF features with the other numerical features
X = data.drop(['commentaires', 'churn'], axis=1).values
X = np.hstack((X, tfidf_features))
y = data['churn'].values

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.5)

# Create a prediction model
model = RandomForestClassifier()
model.fit(X_train, y_train)
# Save the trained model to a file
joblib.dump(model, 'churn_model.pkl')

# Evaluate the model
predictions = model.predict(X_test)
accuracy = accuracy_score(y_test, predictions)
print(f"Model Accuracy: {accuracy}")
the data containes some informations from which the machine learning model learns , to be able to predict the churn for new clients , the problem is that i am not sure that what i did is what really should be done i mean i am not sure the model uses all the column informations to learn and i have created this code to test it on some random data but the predictions it gaves aren't logique , 
import csv
import pandas as pd
import numpy as np
import joblib
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score

# Load the model
model = joblib.load('churn_model.pkl')

# Load the original training data
train_data = pd.read_csv('data.csv')

# Define the new data
new_data = [
    (1008, 22, 6, 4, 275, 1, "Good service"),
    (1009, 40, 9, 11, 500, 0, "Highly satisfied"),
    (1010, 28, 3.5, 7, 200, 2, "Average experience"),
    (1011, 35, 12, 15, 750, 0, "Excellent service"),
    (1012, 30, 8, 10, 300, 3, "Needs improvement"),
    (1013, 55, 2, 1, 50, 0, "Outstanding"),
    (1014, 25, 7, 8, 150, 1, "Satisfied customer"),
    (1015, 45, 15, 12, 650, 2, "Average"),
    (1016, 33, 5, 6, 250, 1, "Good experience"),
    (1017, 38, 10, 9, 350, 0, "Highly dissatisfied")
]

# Create a DataFrame from the new data
column_names = ["numero-du-client", "age", "duree-dengagement-en-semaine", "nombre-de-transactions", "le-prix-d-achat-le-plus-eleve-en-euro", "plaintes-precedents", "commentaires"]
X_new = pd.DataFrame(new_data, columns=column_names)

# Define and fit the TF-IDF vectorizer using the training data
tfidf_vectorizer = TfidfVectorizer()
tfidf_vectorizer.fit(train_data['commentaires'])  # Fit with your training data's comments
tfidf_features = tfidf_vectorizer.transform(X_new['commentaires']).toarray()

# Combine the TF-IDF features with the other numerical features
X_new = X_new.drop('commentaires', axis=1).values
X_new = np.hstack((X_new, tfidf_features))

# Make predictions
predictions = model.predict(X_new)
print("Predicted churn values:", predictions)

can someone help me please ?
